pootle/apps/pootle_statistics/migrations/0016_infer_sub_revisions.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-02-26 13:09
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def infer_submission_revisions(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    revisions = subs.filter(revision__gt=0).values_list(
+        "creation_time", "unit_id", "revision")
+    updated = 0
+    count = 0
+    total = revisions.count()
+    start = time.time()
+    for creation_time, unit_id, revision in revisions.iterator():
```

Comment:
If `revisions` is grouped by `unit` and sorted by `revision` then we can pick the lowest revision to update all the submissions without revision for that unit.